File: android/src/main/java/dev/barooni/capacitor/calendar/models/inputs/DeleteEventsByIdInput.kt
package dev.barooni.capacitor.calendar.models.inputs

import com.getcapacitor.PluginCall
import dev.barooni.capacitor.calendar.PluginError
import dev.barooni.capacitor.calendar.models.enums.EventSpan
import dev.barooni.capacitor.calendar.utils.ImplementationHelper

data class DeleteEventsByIdInput(
    private val call: PluginCall,
) {
    var ids = call.getArray("ids")?.let { ImplementationHelper.jsArrayToLongArray(it) } ?: throw PluginError.MissingId
    val span: EventSpan = call.getInt("span")?.let { EventSpan.fromInt(it) } ?: EventSpan.THIS_EVENT
}

File: ios/Plugin/Models/Inputs/DeleteEventsByIdInput.swift
import Capacitor

struct DeleteEventsByIdInput {
    private let commit: Bool
    private let ids: [String]
    private let span: EventSpan

    init(call: CAPPluginCall) throws {
        self.commit = call.getBool("commit", true)
        guard let ids = call.getArray("ids") as? [String] else {
            throw PluginError.idMissing
        }
        self.ids = ids
        if let spanInt = call.getInt("span"), let span = EventSpan(rawValue: spanInt) {
            self.span = span
        } else {
            self.span = .thisEvent
        }
    }

    func getCommit() -> Bool {
        return commit
    }

    func getIds() -> [String] {
        return ids
    }

    func getSpan() -> EventSpan {
        return span
    }
}
